utilities/mas-imu-test/src/MasImuTest.cpp

```diff
@@ -0,0 +1,1361 @@
+    m_rotationFromEncodersData.reserve(m_commonDataPtr->maxSamples);
+    m_jointsPositionData.reserve(m_commonDataPtr->maxSamples);
+    m_rpyImuData.reserve(m_commonDataPtr->maxSamples);
+    m_rpyRemappedData.reserve(m_commonDataPtr->maxSamples);
+    m_gyroData.reserve(m_commonDataPtr->maxSamples);
+    m_accData.reserve(m_commonDataPtr->maxSamples);
+}
+
+void MasImuTest::MasImuData::clearData()
+{
+    m_errorData.clear();
+    m_rotationFeedbackData.clear();
+    m_rotationFeedbackInInertialData.clear();
+    m_rotationFeedbackInInertialYawFilteredData.clear();
+    m_rotationFromEncodersData.clear();
+    m_jointsPositionData.clear();
+    m_rpyImuData.clear();
+    m_gyroData.clear();
+    m_accData.clear();
+    m_rpyRemappedData.clear();
+}
+
+bool MasImuTest::MasImuData::setupModel()
+{
+    std::string errorPrefix = "[MasImuTest::MasImuData::setupModel](" + m_testName +") ";
+
+    bool ok = m_group->getParameter("imu_frame", m_frameName);
+    if (!ok)
+    {
+        yError() << errorPrefix << "Setup failed.";
+        return false;
+    }
+
+    m_frame = m_commonDataPtr->fullModel.getFrameIndex(m_frameName);
+
+    if (m_frame == iDynTree::FRAME_INVALID_INDEX)
+    {
+        yError() << errorPrefix << "The frame " << m_frameName << " does not exists in the robot model."
+                 << ". Configuration failed.";
+        return false;
+    }
+
+    m_link = m_commonDataPtr->fullModel.getFrameLink(m_frame);
+    assert(m_link != iDynTree::LINK_INVALID_INDEX);
+
+    m_consideredJointIndexes.clear();
+    m_consideredJointNames.clear();
+
+    iDynTree::LinkIndex baseLinkIndex = m_commonDataPtr->traversal.getBaseLink()->getIndex();
+    iDynTree::LinkIndex currentLink = m_link;
+    while (currentLink != baseLinkIndex) {
+        const iDynTree::IJoint* joint = m_commonDataPtr->traversal.getParentJointFromLinkIndex(currentLink);
+        assert(joint);
+        if (joint->getNrOfDOFs() > 0)
+        {
+            m_consideredJointIndexes.push_back(joint->getIndex());
+            m_consideredJointNames.push_back(m_commonDataPtr->fullModel.getJointName(m_consideredJointIndexes.back()));
+        }
+        currentLink = m_commonDataPtr->traversal.getParentLinkFromLinkIndex(currentLink)->getIndex();
+    }
+
+    iDynTree::ModelLoader  reducedModelLoader;
+    ok = reducedModelLoader.loadReducedModelFromFullModel(m_commonDataPtr->fullModel, m_consideredJointNames);
+
+    if (!ok)
+    {
+        yError() << errorPrefix << "Failed to build the reduced model. Configuration failed.";
+        return false;
+    }
+
+    ok = m_kinDyn.loadRobotModel(reducedModelLoader.model());
+
+    m_frame = m_kinDyn.getFrameIndex(m_frameName);
+
+    if (m_frame == iDynTree::FRAME_INVALID_INDEX)
+    {
+        yError() << errorPrefix << "The frame " << m_frameName << " does not exists in the reduced robot model."
+                 << ". Configuration failed.";
+        return false;
+    }
```

Comment:
I suppose these lines exist purposefully to check the frame in the reduced model? After doing similar check in the full model in Lns 68-75?

Reply:
Good point. Removed the check in https://github.com/dic-iit/bipedal-locomotion-framework/pull/62/commits/1e2e7611b1ec7b0d88989d6443a0ff4dad2be274